signature_validator/validate_test.go

```diff
@@ -0,0 +1,76 @@
+package signature_validator
+
+import (
+	"encoding/hex"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+func TestValidateSignature_Init(t *testing.T) {
+
+	type tt = struct {
+		In    string
+		Error error
+	}
+
+	suite := []tt{
+		{"", errors.New("empty string in init")},
+		{"SomeJunk", errors.New("non existent")},
+		{"MasherySha256", nil},
+		{"MasheryMd5", nil},
+	}
+
+	for _, s := range suite {
+
+		validator := SignatureValidator{}
+		err := validator.Init(s.In)
+
+		if err != nil && s.Error == nil {
+			t.Errorf("expected success, got error %s", err.Error())
+			t.FailNow()
+		}
+
+		if err == nil && s.Error != nil {
+			t.Errorf("expected error (%s), got success", s.Error.Error())
+			t.FailNow()
+		}
+	}
+}
+
+func TestValidateSignature_Validate(t *testing.T) {
```

Comment:
Dont we need an end-to-end test with session check as well? The code that pulls the shared secret from the `session_meta_key` should be tested.